### DISCUSSION TITLE
Which component depends on glibc and glibcxx and how we decide which version is the minimum requirement

### DISCUSSION THREAD

**Discussion by ericzhucode** [coder/code-server#5910] [Q&A]
![image](https://user-images.githubusercontent.com/85340915/208693132-d6b1b954-6071-4528-9317-de35c5280c2b.png)
From code server official website we can know that the minimum requirement on Linux is glibc >= 2.17 and glibcxx >= v3.4.18.
1. I want to ask you team what components of vscode core exactly rely on glibc and glibcxx especially for glibc as I don't know why there should be a version requirement of it.
2. For glibcxx, I know it is required to ensure some C++ components like ripgrep/node-pty/spdlog... running properly on target machine or we will get error message like `...glibcxx 3.4.18 does not exist...` but how you detemine that 3.4.18 is the minimum requirement? Since there're multiple C++ components, do we need to check all of them before we decide the required version of glibcxx?

I know it should be a question to vscode team as glibc and glibcxx is required by vscode core but not code-server. However, I would be appreciated it if you team can give me some advice. Thank you!

**Comment by jsjoeio** [comment]
I was trying to remember where that comes from. Some of our requirements come from VS Code's Wiki [here](https://github.com/microsoft/vscode/wiki/How-to-Contribute).

It looks like we added that to the docs here: https://github.com/coder/code-server/pull/1701

And there are some more details about the glibcxx stuff here: https://github.com/coder/code-server/issues/1706

@code-asher may have more information though.

**Comment by code-asher** [comment]
I run `strings` or `objdump` on all the *.node files to figure this out.  I actually get `GLIBCXX_3.4.19` and `GLIBC_2.14` as the minimums right now so I am not sure if we messed something up in the build.

That said, your comment about not knowing why there should be a version requirement for glibc intrigues me.  I am not actually really sure what is going on behind the scenes here.  Do we only need to list glibcxx?  Is the glibc version just the version that was used to compile and glibcxx is what you need to run or something like that?

**Reply by code-asher** [reply]
I should also mention we were inspired by VS Code's requirements to run which mention both glibc and glibcxx: https://code.visualstudio.com/docs/supporting/requirements#_additional-linux-requirements

**Reply by ericzhucode** [reply]
Thank you for you information. Yes I have also noticed this requirement in vscode website. The problem we are facing is that if we build code-server with a newer version image containing GLIBC 2.27 and GLIBCXX 3.4.25, then we will find user whose machine with an older version of GLIBC and GLIBCXX can't meet the requirement to use some functions like terminal or ripgrep search.
I have also found that it seems code-server has this kind of issue before
https://github.com/coder/code-server/pull/1761
From here we can know you team downgraded build image to CentOS7 at that time to limit the GLIBC and GLIBCXX version during linking. Maybe we should try this solution too.

**Reply by code-asher** [reply]
Ah I see!  Yeah I am not sure if there is a way to specifically target a version so the easiest way is to just build with the oldest version of libc you can.
